### DISCUSSION TITLE
Plugin gets fully loaded package from WithContextSetter

### DISCUSSION THREAD

**Discussion by punchio** [golangci/golangci-lint#5214] [Q&A] [answered]
I need to analyze the relationships between functions across multiple packages. Therefore, all packages must be fully loaded during initialization. The initialization should be done in WithContextSetter, so that Run can be executed afterward to verify whether the functions in each package are valid. Is there a way to achieve this?

**Comment by ldez** [comment]
golangci-lint runs by package so you cannot have all the packages.

**Reply by punchio** [reply]
I am creating a check for read-only variables (starting with "ro"). Read-only variables can only be initialized with other read-only variables. They can also be function return values. Therefore, I need to collect all functions first, determine their read-only properties, and then perform read-only checks at each function call site. Can I collect the packages first and perform the check in the last package？

**Reply by ldez** [reply, marked as answer]
No, as I previously explained.

In all cases, your [linter](https://github.com/punchio/go-readonly) will never be accepted because it's based on your personal conventions and name conventions.

**Reply by punchio** [reply]
Thanks, it’s indeed not suitable. I’m planning to write a separate tool.
